### PR TITLE
feat: `cast decode-eof` & `forge inspect <contract> eof`

### PR DESCRIPTION
**Pull request by klkvr**
## Motivation

Adds helpers for decoding and pretty-printing EOF.

Example output:
image

### CODE REVIEW

File: crates/cast/src/lib.rs [hunk 1]
```diff
@@ -1990,6 +1991,24 @@ impl SimpleCast {
...
+    ///
+    /// ```
+    /// use cast::SimpleCast as Cast;
+    ///
+    /// let eof = "0xef0001010004020001005604002000008000046080806040526004361015e100035f80fd5f3560e01c63773d45e01415e1ffee6040600319360112e10028600435906024358201809211e100066020918152f3634e487b7160e01b5f52601160045260245ffd5f80fd0000000000000000000000000124189fc71496f8660db5189f296055ed757632";
+    /// let decoded = Cast::decode_eof(&eof)?;
+    /// println!("{}", decoded);
+    /// # Ok::<(), eyre::Report>(())
+    pub fn decode_eof(eof: &str) -> Result<String> {
+        let eof_hex = hex::decode(strip_0x(eof))?;
```
Comment (DaniPopes) [inline crates/cast/src/lib.rs:2008]: doesnt need stripping

File: crates/common/fmt/src/eof.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            table.add_row(vec![&idx.to_string(), &container.to_string()]);
+        }
+
+        write!(result, "\n\nContainer sections:\n{table}")?;
+    }
+
+    if !data_section.is_empty() {
+        let mut table = Table::new();
+        table.set_content_arrangement(ContentArrangement::Dynamic);
+        table.add_row(vec![&data_section.to_string()]);
```
Comment (DaniPopes) [inline crates/common/fmt/src/eof.rs:70]: could it be worth chunking in 32 bytes?

File: crates/forge/bin/cmd/inspect.rs [hunk 3]
```diff
@@ -300,6 +310,8 @@ impl_value_enum! {
         Ewasm             => "ewasm" | "e-wasm",
         Errors            => "errors" | "er",
         Events            => "events" | "ev",
+        Eof => "eof" | "eof-container" | "eof-deployed",
+        EofInit => "eof-init" | "eof-initcode" | "eof-initcontainer",
```
Comment (DaniPopes) [inline crates/forge/bin/cmd/inspect.rs:314]: align

**Review by DaniPopes — APPROVED** [review]

File: crates/common/fmt/src/eof.rs (hunk 2 above)
Comment (klkvr) [reply crates/common/fmt/src/eof.rs:70]: not sure, currently solc is putting metadata at the beginning so the actual 32-bytes values aren't aligned, so such view wouldn't really make it more readable

**Pull request merged**